lib/ansible/modules/system/locale_gen.py

```diff
@@ -103,15 +102,16 @@ def fix_case(name):
 def replace_line(existing_line, new_line):
     """Replaces lines in /etc/locale.gen"""
     try:
-        f = open("/etc/locale.gen", "r")
-        lines = [line.replace(existing_line, new_line) for line in f]
-    finally:
-        f.close()
+        with open("/etc/locale.gen", "r") as f:
+            lines = [line.replace(existing_line, new_line) for line in f.readlines()]
```

Comment:
```suggestion
            lines = [line.replace(existing_line, new_line) for line in f]
```
Matches the original code more closely.